aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,102 @@
 
-**Removed**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet2&highlight=3-7,14-19)]
 
-The item was manually removed by a call to `Remove()`
+The following code calls [Get](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) to fetch the cached time:
 
-**Replaced**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_gct)]
 
-The item was overwritten.
+See [IMemoryCache methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache) and [CacheExtensions methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) for a description of the cache methods.
 
-**Expired**
+## Using MemoryCacheEntryOptions
 
-The item timed out.
+The following sample:
 
-**TokenExpired**
+- Sets the absolute expiration time. This is the maximum time the entry can be cached and prevents the item from becoming too stale when the sliding expiration is continuously renewed.
+- Sets a sliding expiration time. Requests that access this cached item will reset the sliding expiration clock.
+- Sets the cache priority to `CacheItemPriority.NeverRemove`. 
+- Sets a [PostEvictionDelegate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.postevictiondelegate) that will be called after the entry is evicted from the cache. The callback is run on a different thread from the code that removes the item from the cache.
 
-The token the item depended upon fired an event.
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_et)]
 
-**Capacity**
+## Cache dependencies
 
-The item was removed as part of the cache's memory management process.
+The following sample shows how to expire a cache entry if a dependent entry expires. A `CancellationChangeToken` is added to the cached item. When `Cancel` is called on the `CancellationTokenSource`, both cache entries are evicted. 
 
-You can specify that one or more cache entries depend on a `CancellationTokenSource` by adding the expiration token to the `MemoryCacheEntryOptions` object. When a cached item is invalidated, call `Cancel` on the token, which will expire all of the associated cache entries (with a reason of `TokenExpired`). The following unit test demonstrates this:
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_ed)]
```

Comment:
Again the sample here is a little busy. We should at least highlight the relevant calls to adding the entry with a `CancellationChangeToken` and the actual call to `Cancel`